web_external/ClipExplorer/index.jsx

```diff
@@ -0,0 +1,227 @@
+import _ from 'underscore';
+import React, { Component } from 'react';
+import { Modal, ButtonToolbar, Button, Row, Col, Glyphicon, Checkbox } from 'react-bootstrap';
+import { restRequest } from 'girder/rest';
+
+import style from './style.styl';
+
+class ClipExplorer extends Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            paths: [{ name: 'root', folderId: 'root' }],
+            currentFolders: [],
+            rootFolders: null,
+            annotationFolders: new Set(),
+            selectedItem: null,
+            selectedFolderId: null,
+            showMode: 'all'
+        };
+        this.folderCache = new Map();
+    }
+
+    componentDidMount() {
+        this.fetchFolder();
+        this.getAllAnnotationFolders();
+    }
+
+    getAllFoldersAtRoot() {
+        return Promise.all([
+            restRequest({
+                url: '/collection',
+            }).then((collections) => {
+                return Promise.all(collections.map((collection) => {
+                    return restRequest({
+                        url: 'folder',
+                        data: {
+                            parentType: 'collection',
+                            parentId: collection._id
+                        }
+                    })
+                }));
+            })
+        ]).then((results) => {
+            var folders = _.flatten(results);
+            return this.attachImageItem(folders);
+        })
+    }
+
+    getAllAnnotationFolders() {
+        // The current set of API is lacking flexible search, this is the method to get all annotated video clips (folder)
+        restRequest({
+            url: '/item',
+            data: {
```

Comment:
Reminder: this needs to specify 'no item limit', as you realized and noted during testing.

Reply:
Thanks for the reminder.